src/services/replyService.js

```diff
@@ -12,65 +12,75 @@ const {
 }                = require('../utils/constants');
 
 class ReplyService {
-  constructor(postFromOurDb) {
-    this.post = postFromOurDb;
+  constructor({ postId }) {
+    this.postId = postId;
     this.postAuthor;
-    this.parent;
+    this.postPermLink;
   }
```

Comment:
```
constructor({ postId }) {
   this.postId = postId;
}

fetchAllPostReplies() {
   return PostModel.findById(this.postId)
   ...
}
```

A service shouldn't receive an individual id and set it as a class variable. It shouldn't hold single individual models or model attributes (ie: `this.postAuthor, this.postPermLink`).

The reason for this is Services are lightweight modules that hold functions used in multiple areas in the codebase. You simply import the service and use its helper functions. To keep service files from becoming too large, we separate each service by the model it's associated with. 

So the `fetchAllPotsReplies` method should simply receive a `postId` as params:
```
fetchAllPostReplies(postId) {
   return PostModel.findById(postId)
   ...
}
```

This way, we can simply import the service and use the `fetchAllPostReplies` method whenever/wherever we want and fetch whichever post's replies we want. Currently, we have to instantiate a brand new reply service every time we want to use the `fetchAllPostReplies` function.

Think I mentioned this in a previous pull request. Maybe I didn't explain well enough

Reply:
Yes I understood.  This amkes sense to code this way.  I have commented several times that I will get to this change.  I will get to this change when I have a chance.  Sorry, I have been busy on another contract.

Reply:
For sure. I didn't see the comments you made. My bad